### PR TITLE
Move experimental-features to settings in nix.nix

### PR DESCRIPTION
**Pull request by 497a**

### CODE REVIEW

File: modules/common/nix.nix [hunk 1]
```diff
@@ -1,10 +1,8 @@
 { lib, pkgs, ... }: {
   nix = {
     # enable flakes
-    extraOptions = lib.mkDefault ''
-      experimental-features = nix-command flakes
-    '';
     settings = {
+      experimental-features = [ "nix-command" "flakes" ];
```
Comment (peanutbother) [inline modules/common/nix.nix]: add `lib.mkDefault` or similar to allow overriding this setting

**Review by peanutbother — CHANGES_REQUESTED** [review]
thanks for the contribution!
I first have to test the effects / read on the docs as there might be semantic differences.
In a way to allow to override this parameter `lib.mkDefault` was used in the first place.
There exists a lib function which does the same for arrays but I have to look that up too, maybe you could read up on this?

**Comment by 497a** [comment]
Done!

**Comment by 497a** [comment]
I built your config locally without any issues before committing.

**Review by peanutbother — APPROVED** [review]

**Pull request merged**